add Japanese translation

a1678991 opened the pull request.

a1678991 committed:
add Japanese translation

a1678991 committed:
fix

kjen93 commented on Japanese/JapaneseAutoUpdate.rc:
```diff
@@ -0,0 +1,16 @@
+STRINGTABLE
+LANGUAGE LANG_JAPANESE, SUBLANG_JAPANESE_JP
```
It seems like we should use
`LANGUAGE LANG_JAPANESE, SUBLANG_DEFAULT`
instead of SUBLANG_JAPANESE_JP

Could you please update resource files to use the correct header, so I can merge the changes into the correct branch?

kjen93 commented on Japanese/JapaneseInstaller.iss:
```diff
@@ -0,0 +1,29 @@
+[CustomMessages]
```
Inno Setup requires localization files to be saved using **UTF-8 with BOM**.
Could you please add BOM to this file, so I can merge this?
There is no need to add BOM to other files.

kjen93 reviewed (changes requested):
Could you please make some changes I've described in comments so I can merge this?
I don't have write access to your fork, so I can't make these changes myself

a1678991 committed:
fix LANGUAGE

a1678991 committed:
add BOM

kjen93 commented:
Thank you for the translation!

I've prepared beta version of Battery Mode with this Pull Request included, which can be downloaded here: https://sdl.tarcode.ru/beta/3.8.9.111/

Could you please install it and check that translation has been applied correctly?
I would recommend checking the following things:

- [ ] Japanese language is suggested automatically when starting the installer on Japanese version of Windows
- [ ] Japanese translation is correctly applied to the installer
- [ ] Japanese language is used as a default language on Japanese version of Windows
- [ ] Japanese translation is correctly applied to "Update check" dialogs
- [ ] Japanese translation is correctly applied to "Autorun" dialogs
- [ ] Japanese translation is correctly applied to Battery Mode main window
- [ ] Japanese translation is correctly applied to Battery Mode tray menu
- [ ] Japanese translation is correctly applied to Battery Mode tray icon's mouseover popup
- [ ] Japanese translation is correctly applied to hotkey configuration window
- [ ] Japanese translation is correctly applied to power system information window
- [ ] Japanese translation is correctly applied to scheduler
- [ ] Japanese translation is correctly applied to Menu Access Keys

In case any problems are found during the testing, please feel free to make additional changes and request another beta build.

a1678991 committed:
update translation

a1678991 committed:
upadte translation

a1678991 committed:
fix

a1678991 commented:
BatteryModeInstaller.exe
Japanese is automatically selected on any computer.
BatteryModeInstaller64.exe
English is selected on computers that have never been installed this version.
Once installed in Japanese even after uninstalling the default language will be set to Japanese.

a1678991 commented:
The Japanese of Language seems to be wrong.
![image](https://user-images.githubusercontent.com/7441512/34650570-90931e02-f406-11e7-9ac2-d69323d19b3a.png)

a1678991 committed:
update translation

a1678991 committed:
fix translation

a1678991 commented:
The priority is not high, but the font of the name of the rule is not suitable for Japanese, making it difficult to read "kanji".
I think it will improve if you change to another font or increase the size.
![image](https://user-images.githubusercontent.com/7441512/34650496-6b853f06-f405-11e7-93ed-4fd4feb4bc90.png)

a1678991 commented:
- [x] Japanese language is suggested automatically when starting the installer on Japanese version of Windows
- [x] Japanese translation is correctly applied to the installer
- [x] Japanese language is used as a default language on Japanese version of Windows
- [x] Japanese translation is correctly applied to "Update check" dialogs
- [x] Japanese translation is correctly applied to "Autorun" dialogs
- [x] Japanese translation is correctly applied to Battery Mode main window
- [x] Japanese translation is correctly applied to Battery Mode tray menu
- [x] Japanese translation is correctly applied to Battery Mode tray icon's mouseover popup
- [x] Japanese translation is correctly applied to hotkey configuration window
- [x] Japanese translation is correctly applied to power system information window
- [x] Japanese translation is correctly applied to scheduler
- [x] Japanese translation is correctly applied to Menu Access Keys

kjen93 commented:
Thank you for testing and updates!

> BatteryModeInstaller64.exe
> English is selected on computers that have never been installed this version.
> Once installed in Japanese even after uninstalling the default language will be set to Japanese.

I think this happens because installer language preference is saved after successful installation and kept until you uninstall the app.
Could you please test this in the following way:
1. Uninstall 64-bit version from Control Panel (and make sure that installation folder has been deleted)
2. Launch beta version's 64-bit installer (it should suggest Japanese)

> The Japanese of Language seems to be wrong.

(X) part was added due to Menu Access Key autogeneration, so I think it works as intended.

> The priority is not high, but the font of the name of the rule is not suitable for Japanese, making it difficult to read "kanji".
> I think it will improve if you change to another font or increase the size.

I've changed the font in new beta build. Could you please try it and tell me if this looks better?

You can download new beta build here: https://sdl.tarcode.ru/beta/3.8.9.111/

Also, I performed some testing myself and the following places seemed strange to me. Could you please take a look at these and fix them (or tell me that it is expected)?

1. Color scheme's Menu Access Keys are not mapped correctly:
![screen shot 2018-01-07 at 20 33 52](https://user-images.githubusercontent.com/156758/34652201-4bb4e8ca-f3ec-11e7-80b1-0e603c59c548.png)

2. Menu Access Key for the last entry is duplicated:
![screen shot 2018-01-07 at 20 34 49](https://user-images.githubusercontent.com/156758/34652219-66cbc2b4-f3ec-11e7-9d49-ff68ab458b97.png)

3. (D) Menu Access Key is not mapped correctly
![screen shot 2018-01-07 at 20 35 24](https://user-images.githubusercontent.com/156758/34652233-8b920c48-f3ec-11e7-9f96-34125d26953e.png)

4. "If" and "While" are not translated
![screen shot 2018-01-07 at 20 37 16](https://user-images.githubusercontent.com/156758/34652241-a87b0d50-f3ec-11e7-8d20-ef7a1c46ff68.png)

5. Menu Access Keys in Power System Information window are duplicated
![screen shot 2018-01-07 at 20 38 52](https://user-images.githubusercontent.com/156758/34652248-c1f45f66-f3ec-11e7-8b65-28d3f83ee6bf.png)

a1678991 committed:
fix translation

a1678991 committed:
update translation

a1678991 committed:
fix translation

a1678991 committed:
update translation

a1678991 commented:
I understood the language selection of the installer.
>(X) part was added due to Menu Access Key autogeneration, so I think it works as intended.

Changed to (J) appropriate for Japanese.
>I've changed the font in new beta build. Could you please try it and tell me if this looks better?

There is no problem with this font at all.
>1. Color scheme's Menu Access Keys are not mapped correctly:

Fixed a39db3e
>2. Menu Access Key for the last entry is duplicated:

Fixed b27a863
>3. (D) Menu Access Key is not mapped correctly

I do not know the solution to this problem.
But I changed the language menu to English.
It may works.
>4. "If" and "While" are not translated

I think that people who use this function are easier to use if and while than written in Japanese.
>5. Menu Access Keys in Power System Information window are duplicated

Fixed c3b1c73

a1678991 committed:
fix translation

a1678991 committed:
fix translation

a1678991 committed:
update translation

a1678991 committed:
update translation

kjen93 commented:
Thank you for the fixes!

> I do not know the solution to this problem.
> But I changed the language menu to English.
> It may works.

I'll merge this Pull Request and fix this issue myself. Please create another Pull Request if you need to make additional changes.

kjen93 commented:
Beta version https://sdl.tarcode.ru/beta/3.8.9.111/ has been updated with this Pull Request.
My fixes were also included in this build.
I'll release an update for Battery Mode application soon if no new issues are found.

Thank you!
